code-comment-style.md

```diff
@@ -0,0 +1,137 @@
+# Code Comment Style
+
+This document describes the code comment style applied to TiKV repositories. When you are to commit, be sure to follow the style to write good code comments.
+
+## Why a good comment matters?
+
+- To speed up the reviewing process
+- To help maintain the code
+- To improve the API document readability
+- To improve the development efficiency of the whole team
+
+## Where/When to comment?
+
+Write a comment where/when:
+
+- For important code
+- For obscure code
+- For tricky or interesting code
+- For a complex code block
+- If a bug exists in the code but you cannot fix it or you just want to ignore it for the moment
+- If the code is not optimal but you don’t have a smarter way now
+- To remind yourself or others of missing functionality or upcoming requirements not present in the code
+
+A comment is generally required for:
+
+- File
+- Type
+- Constant 
+- Function
+- Method
+- Variable
+- Typical algorithm
+- Exported name
+- Test case
+- TODO
+- FIXME
+
+## How to comment?
+
+### Format of a good comment (for Rust)
+
+- Non-doc comment

```

Comment:
When to use non-doc comment and when not to use?

Reply:
As a general thumb of rule, use doc comment as long as you want to refer to that information in the command line. In our case, we have the convention to use doc comment for (public or private) structures including the fields, macros, methods, etc.  We use non-doc (code) comment for everything else like inner steps of functions.

Will add this in the comment style guide.

Reply:
I would state the distinction as "use doc comments to document the interface of code and use non-doc comments to document the implementation details"